src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -98,6 +95,14 @@ public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
         SCMRevision rev = scmSource.getTrustedRevision(tip, listener);
         GitSCM gitSCM = (GitSCM) scmSource.build(head, rev);
 
+        this.gitConfig = new GitConfig();
+        this.gitConfig.setGitBranch(property.getBranch().getName());
+
+        if(gitConfig.getGitBranch().contains("PR")){
+            String[] refSpecs = gitSCM.getUserRemoteConfigs().get(0).getRefspec().split("/", 0);
+            gitConfig.setGitBranch(refSpecs[refSpecs.length - 1]);
+        }
```

Comment:
To get the branch name I was using ```property.getBranch().getName()```, but if the the build will correspond to a PR then ```property.getBranch().getName()``` gives ```PR-#PRnumber```. Hence after some time I got the branch name like this. Not sure if it is the best way.

Reply:
I think it might be better to check if the name of the starts with "PR-", rather than `.contains("PR"))` 

and it might be problematic to split one on `/` in case the branch name contains `/`

Reply:
Agreed. This will require some heuristics at some point. PR branch names on different services are different. Github, Bitbucket, etc. all have their own way to naming their PR branches.

Reply:
Perhaps this could even be an extension point? Maybe SPRP could try to fetch different known refs until one of them works, this would make life easy and simple for the user. Then it could memorize which one worked for the next time.

Reply:
I have moved this to the second phase of coding and Jira for this is https://issues.jenkins-ci.org/browse/JENKINS-51867